sklearn/feature_extraction/_dict_vectorizer.py

```diff
@@ -98,6 +121,32 @@ def __init__(self, *, dtype=np.float64, separator="=", sparse=True,
         self.sparse = sparse
         self.sort = sort
 
+    def _add_element(self, f, v, feature_names, vocab, fitting=True,
+                     transforming=False, indices=None, values=None):
```

Comment:
This method does a bunch given its name of `_add_element`. If there is not a way to simplify this, I think a docstring would be helpful for maintenance.

Reply:
I've simplified the function: some code is duplicated, but the function is now called `_add_iterable_element` and used only in case of iterable elements. Anyway, I have also added a small docstring.